src/maliput_osm/osm/osm_manager.h

```diff
@@ -63,6 +65,36 @@ class OSMManager {
   const std::unordered_map<Segment::Id, Segment>& GetOSMSegments() const;
 
  private:
+  // Convenient definitions for the AddLanesToSegment method's left parameter.
+  // @{
+  static constexpr bool kAdjacentLeft{true};
+  static constexpr bool kAdjacentRight{!kAdjacentLeft};
+  // @}
+
+  // Recurrent method for finding the left or right adjacent lanes of a given @p lane_id and populate the @p segment.
+  //
+  // @param lane_id The lane_id to find the adjacent lanes of.
+  // @param lanes The lanes to search in.
+  // @param left Whether to search for the left or right adjacent lanes.
+  // @param[out] segment The segment to be filled. The lanes vector is filled from rightest most lane to leftest most
+  // lane.
+  //
+  // @throws maliput::common::assertion_error When the adjacent lane id doesn't match any lane in the @p lanes.
+  static void AddLanesToSegment(const Lane::Id& lane_id, const std::unordered_map<Lane::Id, Lane>& lanes, bool left,
+                                Segment& segment);
```

Comment:
`Segment& segment` should be `Segment* segment` due to our styleguide.

Reply:
Done.